src/SFA.DAS.EAS.Web/ViewModels/TransactionsDownloadResultViewModel.cs

```diff
@@ -1,7 +1,131 @@
-﻿namespace SFA.DAS.EAS.Web.ViewModels
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using SFA.DAS.EAS.Domain.Data.Entities.Account;
+using SFA.DAS.EAS.Domain.Models.Transaction;
+
+namespace SFA.DAS.EAS.Web.ViewModels
 {
     public class TransactionsDownloadResultViewModel
     {
-        public string HashedAccountId { get; set; }
+        public Account Account { get; set; }
+
+        [Range(1, 12)]
+        public int FromMonth { get; set; }
+
+
+        [Range(1, 9999)]
+        public int FromYear { get; set; }
+
+
+        [Range(1, 12)]
+        public int ToMonth { get; set; }
+
+
+        [Range(1, 9999)]
+        public int ToYear { get; set; }
+
+        public bool Valid => ValidFromDate && ValidToDate && !FromDateInFuture && !ToDateInFuture;
+
+        public bool ValidFromDate
+        {
+            get
+            {
+                if (!ValidFromMonth()) return false;
+                if (!ValidFromYear()) return false;
+
+                return true;
+            }
+        }
+
+        public bool ValidToDate
+        {
+            get
+            {
+                if (!ValidToMonth()) return false;
+                if (!ValidToYear()) return false;
+
+                return true;
+            }
+        }
+
+        public bool FromDateInFuture => ValidFromDate && DateInFuture(FromMonth, FromYear);
+
+        public bool ToDateInFuture => ValidToDate && DateInFuture(ToMonth, ToYear);
+
+        public DateTime FromDate()
+        {
+            if(!ValidFromMonth())
+                throw new ArgumentException("This object has an invalid FromMonth");
+
+            if (!ValidFromYear())
+                throw new ArgumentException("This object has an invalid FromYear");
+
+            if (FromDateInFuture)
+                throw new ArgumentException("This FromYear is in the future");
+
+            return new DateTime(FromYear, FromMonth, 1);
+        }
+
+        public DateTime ToDate()
```

Comment:
ToDate doesn't make it clear what is going on - it is the last second of the month that is being returned. Would it better to specify millisecond as well?

Reply:
Given how the transactions work I doubt we need to go down to that level of accuracy. i'll update the method name though

Reply:
Milli- seconds would not be good, but this is edge case